FIX #3349 Checking a VAT number with spaces

tomours opened the pull request:
This will allow VAT check to work properly even if VAT has been typed with spaces in.

tomours committed:
FIX #3349 Allow checking a VAT number with spaces

One of the first things I did, while starting to try Dolibarr, was to fill info about my company.
I filled my VAT number : "FR 13 794..." and saw the "check" button.
Oh, that's nice ! Click. Didn't work :(
In fact, it didn't work because the webservice Dolibarr uses to check VAT numbers doesn't understand spaces.

By doing this we could enter good looking VAT numbers (with spaces) and still check them.

rdoursenaud commented:
Please squash the two commits into one.
See: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits

And please follow the commit naming guideline at https://github.com/Dolibarr/dolibarr/blob/develop/CONTRIBUTING.md#commits

Something like:

```
CLOSE #3349 Checking a VAT number with spaces

[Long description]
```

Thanks!

tomours commented:
Well, sorry, I tried but I think I made it worse... there is now 4 commit lines in the pull request and I don't know how to remove them (I'm not completely sure that I'm understanding what I'm doing).

rdoursenaud commented:
Failing is learning. Keep trying ;)

Let me give you a hand. Here's how I'd do it:

```
git rebase -i f3a2bff~  # Rebase from the commit before your first
```

I'd change the rebasing rule to

```
pick f3a2bff Checking a VAT number with spaces
squash 9c2b970 Checking a VAT number with spaces - catch any type of whitespace
```

removing everything else.

Here `squash` asks us for a common commit message.
I'd grab the opportunity to make the commit message meet the criterias.
Something like:

```
FIX #3349 Allow checking a VAT number with spaces

One of the first things I did, while starting to try Dolibarr was to fill info about my company.
I filled my VAT number : "FR 13 794..." and saw the "check" button.
Oh, that's nice ! Click. Didn't work :(
In fact, it didn't work because the webservice Dolibarr uses to check VAT numbers
doesn't understand spaces.

By doing this we could enter good looking VAT numbers (with spaces) and still check them.
```

Now, just push to your repository.
We need to use `--force` because we have rewritten history.

```
git push --force
```

Done :)

tomours commented:
Oh my god, it worked ! :+1: 
I think I may have been close at some time, but was unable to push (without force option), and that's where I started my mess.
Well thanks a lot, I think I'm starting to understand the big lines ! :)

rdoursenaud commented:
:+1: 

You're welcome :)

tomours committed:
FIX  #3372 - Printing issue with Postgres and dates (SQL format)

After connecting to the database, we force datestyle to "ISO, YMD", so dates returned by the db will have a format compatible with Dolibarr functions.
This will fix dates printing issues in mostly all pages, with a postgres db configured with "SQL, DMY" as default.
